### PR TITLE
Allow setting Subject Name and Subject Alternative Name

### LINKED ISSUE #1257: Record both domain and IP in node certificate SAN
We currently only support either the node's RPC IP address or domain (if `--domain` is passed to `cchost`) in the Subject Alternative Name of the node's x509 certificate. 

It might be useful to support both IP and DNS in there so that an operator can connect to a node _securely_ without relying on the DNS/load balancer to to be up and running.

### PR DESCRIPTION
**Pull request by achamayou**
fix #1257

### CODE REVIEW

File: src/host/main.cpp [hunk 1]
```diff
@@ -297,6 +297,16 @@ int main(int argc, char** argv)
   app.add_option(
     "--domain", domain, "DNS to use for TLS certificate validation");
 
+  std::string subject_name("CN=CCF Node");
+  app.add_option("--sn", subject_name, "Subject Name in node certificate, eg. CN=CCF Node")
+    ->capture_default_str();
+
+  std::vector<std::string> subject_alternative_names;
+  app.add_option(
```
Comment (jumaffre) [inline src/host/main.cpp:305]: Should we use a special parser here, similarly to the address options, to check that the values are in the expected format?

Comment (achamayou) [reply src/host/main.cpp:305]: I hadn't thought of that, but absolutely, yes!

File: src/ds/cli_helper.h [hunk 2]
```diff
@@ -2,8 +2,13 @@
 // Licensed under the Apache 2.0 License.
 #pragma once
 
+#include "../tls/san.h"
```
Comment (jumaffre) [inline src/ds/cli_helper.h:5]: Minor: `#include "tls/san.h`

File: src/node/node_state.h [hunk 3]
```diff
@@ -67,6 +67,9 @@ namespace ccf
   using RaftType =
     raft::Raft<consensus::LedgerEnclave, NodeToNode, Snapshotter>;
 
+  static const std::string IP_ADDRESS_PREFIX("iPAddress:");
```
Comment (jumaffre) [inline src/node/node_state.h:70]: Duplicated in `cli_helper.h`.

Comment (achamayou) [reply src/node/node_state.h:70]: Yes, forgot to remove it when it moved, thanks for spotting it.

File: src/ds/cli_helper.h (hunk 2 above)
Comment (achamayou) [reply src/ds/cli_helper.h:5]: Thank you.

**Review by jumaffre — APPROVED** [review]

**Pull request merged**